dask/dataframe/indexing.py

```diff
@@ -13,24 +13,71 @@
 from ..base import tokenize
 
 
-class _LocIndexer(object):
-    """ Helper class for the .loc accessor """
-
+class _IndexerBase(object):
     def __init__(self, obj):
         self.obj = obj
 
     @property
     def _name(self):
         return self.obj._name
 
+    @property
+    def _meta_indexer(self):
+        raise NotImplementedError
+
     def _make_meta(self, iindexer, cindexer):
         """
         get metadata
         """
         if cindexer is None:
             return self.obj
         else:
-            return self.obj._meta.loc[:, cindexer]
+            return self._meta_indexer[:, cindexer]
+
+
+class _iLocIndexer(_IndexerBase):
+
+    @property
+    def _meta_indexer(self):
+        return self.obj._meta.iloc
+
+    def __getitem__(self, key):
+
+        if self.obj.ndim == 1:
+            if key != slice(None):
+                raise ValueError("'Series.iloc' only supports 'slice(None)'.")
+            return self.obj
+
+        # dataframe
+        msg = ("'DataFrame.iloc' does not support slicing rows. "
+               "The indexer must be a 2-tuple whose first item is "
+               "'slice(None)'.")
```

Comment:
I suggest that we explicitly say that it can only be used to select columns, like `.iloc[:, 'col']` rather than specify the technical types that are allowed.

Reply:
Much better, thanks!